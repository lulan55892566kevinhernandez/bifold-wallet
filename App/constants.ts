import { IndyCredentialMetadata } from '@aries-framework/core/build/types'

export const defaultLanguage = 'en'

export enum LocalStorageKeys {
  Onboarding = 'OnboardingState',
}

// FIXME: Remove once fixed in AFJ
export interface IndexedIndyCredentialMetadata extends IndyCredentialMetadata {
  [key: string]: string | undefined
}

export const indyCredentialKey = '_internal/indyCredential'

//Keys for items saved in keychain/async storage
export const KEYCHAIN_SERVICE_KEY = 'walletkey'
export const KEYCHAIN_SERVICE_ID = 'walletid'
export const STORAGE_KEY_SALT = 'savedalt'
export const STORAGE_FIRSTLOGIN = 'firstlogin'
export const STORAGE_AUTHLEVEL = 'authlevel'

export const dateFormatOptions: { year: 'numeric'; month: 'short'; day: 'numeric' } = {
  year: 'numeric',
  month: 'short',
  day: 'numeric',
}