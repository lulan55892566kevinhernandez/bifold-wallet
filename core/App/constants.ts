export const defaultLanguage = 'en'

// Used to property prefix TestIDs so they can be looked up
// by on-device automated testing systems like SauceLabs.
export const testIdPrefix = 'com.ariesbifold:id/'

export enum LocalStorageKeys {
  Onboarding = 'OnboardingState',
  Privacy = 'PrivacyState',
  // FIXME: Once hooks are updated this should no longer be necessary
  RevokedCredentials = 'RevokedCredentials',
  RevokedCredentialsMessageDismissed = 'RevokedCredentialsMessageDismissed',
  Preferences = 'PreferencesState',
}

export enum KeychainServices {
  Salt = 'secret.wallet.salt',
  Key = 'secret.wallet.key',
}

export const walletId = 'walletId'

export const dateFormatOptions: { year: 'numeric'; month: 'short'; day: 'numeric' } = {
  year: 'numeric',
  month: 'short',
  day: 'numeric',
}

export const minPINLength = 6
