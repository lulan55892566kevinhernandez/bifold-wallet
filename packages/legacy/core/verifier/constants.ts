import { PredicateType } from '@aries-framework/core'

import { ProofRequestTemplate, ProofRequestType } from './types/proof-reqeust-template'

export const defaultProofRequestTemplates: Array<ProofRequestTemplate> = [
  {
    id: 'Aries:5:StudentFullName:0.0.1:indy',
    name: 'Student full name',
    description: 'Verify the full name of a student',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'XUxBrVSALWHLeycAUhrNr9:3:CL:26293:Student Card',
          requestedAttributes: [
            {
              name: 'student_first_name',
              restrictions: [{ schema_id: 'XUxBrVSALWHLeycAUhrNr9:3:CL:26293:Student Card' }],
            },
            {
              name: 'student_last_name',
              restrictions: [{ schema_id: 'XUxBrVSALWHLeycAUhrNr9:3:CL:26293:Student Card' }],
            },
          ],
        },
      ],
    },
  },
  {
    id: 'Aries:5:StudentFullNameAndExpirationDate:0.0.1:indy',
    name: 'Student full name and expiration date',
    description: 'Verify that full name of a student and that he/she has a not expired student card.',
    version: '0.0.1',
    payload: {
      type: ProofRequestType.Indy,
      data: [
        {
          schema: 'XUxBrVSALWHLeycAUhrNr9:3:CL:26293:Student Card',
          requestedAttributes: [
            {
              names: ['student_first_name', 'student_last_name'],
              restrictions: [{ schema_id: 'XUxBrVSALWHLeycAUhrNr9:3:CL:26293:Student Card' }],
            },
          ],
          requestedPredicates: [
            {
              name: 'expiry_date',
              predicateType: PredicateType.GreaterThanOrEqualTo,
              predicateValue: 20240101,
              restrictions: [{ schema_id: 'XUxBrVSALWHLeycAUhrNr9:3:CL:26293:Student Card' }],
            },
          ],
        },
      ],
    },
  },
]

export const protocolVersion = 'v1'
export const domain = 'didcomm://invite'
