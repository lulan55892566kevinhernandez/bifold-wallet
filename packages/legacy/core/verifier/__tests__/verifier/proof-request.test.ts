import {
  buildProofRequestDataForTemplate,
  buildProofRequestDataForTemplateId,
  hasPredicates,
  isParameterizable,
} from '../../utils/proof-request'
import SpyInstance = jest.SpyInstance
import { defaultProofRequestTemplates } from '../../constants'

describe('Helpers', () => {
  let spy: SpyInstance

  beforeAll(() => {
    spy = jest.spyOn(Date, 'now').mockImplementation(() => 1677766511505)
  })

  test('Build indy proof request from template containing two requested attributes', async () => {
    const template = defaultProofRequestTemplates[0]
    const proofRequest = buildProofRequestDataForTemplate(template)
    expect(proofRequest).toMatchSnapshot()
  })

  test('Build indy proof request from template containing two requested attributes and predicate', async () => {
    const template = defaultProofRequestTemplates[1]
    const proofRequest = buildProofRequestDataForTemplate(template)
    expect(proofRequest).toMatchSnapshot()
  })

  test('Build indy proof request from template id', async () => {
    const templateId = defaultProofRequestTemplates[0].id
    const proofRequest = buildProofRequestDataForTemplateId(templateId)
    expect(proofRequest).toMatchSnapshot()
  })

  test('Check if proof has predicates', async () => {
    expect(hasPredicates(defaultProofRequestTemplates[0])).toBeFalsy()
    expect(hasPredicates(defaultProofRequestTemplates[1])).toBeTruthy()
  })

  afterAll(() => {
    spy.mockRestore()
  })
})
