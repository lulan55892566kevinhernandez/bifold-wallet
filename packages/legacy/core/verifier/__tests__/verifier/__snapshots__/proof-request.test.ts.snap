// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers Build indy proof request from template containing two requested attributes 1`] = `
Object {
  "indy": Object {
    "name": "Student full name",
    "nonce": "1677766511505",
    "requestedAttributes": Map {
      "referent_0" => ProofAttributeInfo {
        "name": "student_first_name",
        "names": undefined,
        "nonRevoked": undefined,
        "restrictions": Array [
          AttributeFilter {
            "attributeValue": undefined,
            "credentialDefinitionId": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
            "issuerDid": undefined,
            "schemaId": undefined,
            "schemaIssuerDid": undefined,
            "schemaName": undefined,
            "schemaVersion": undefined,
          },
        ],
      },
      "referent_1" => ProofAttributeInfo {
        "name": "student_last_name",
        "names": undefined,
        "nonRevoked": undefined,
        "restrictions": Array [
          AttributeFilter {
            "attributeValue": undefined,
            "credentialDefinitionId": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
            "issuerDid": undefined,
            "schemaId": undefined,
            "schemaIssuerDid": undefined,
            "schemaName": undefined,
            "schemaVersion": undefined,
          },
        ],
      },
    },
    "requestedPredicates": Map {},
    "version": "0.0.1",
  },
}
`;

exports[`Helpers Build indy proof request from template containing two requested attributes and predicate 1`] = `
Object {
  "indy": Object {
    "name": "Student full name and expiration date",
    "nonce": "1677766511505",
    "requestedAttributes": Map {
      "referent_0" => ProofAttributeInfo {
        "name": undefined,
        "names": Array [
          "student_first_name",
          "student_last_name",
        ],
        "nonRevoked": undefined,
        "restrictions": Array [
          AttributeFilter {
            "attributeValue": undefined,
            "credentialDefinitionId": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
            "issuerDid": undefined,
            "schemaId": undefined,
            "schemaIssuerDid": undefined,
            "schemaName": undefined,
            "schemaVersion": undefined,
          },
        ],
      },
    },
    "requestedPredicates": Map {
      "referent_1" => ProofPredicateInfo {
        "name": "expiry_date",
        "nonRevoked": undefined,
        "predicateType": ">=",
        "predicateValue": 20240101,
        "restrictions": Array [
          AttributeFilter {
            "attributeValue": undefined,
            "credentialDefinitionId": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
            "issuerDid": undefined,
            "schemaId": undefined,
            "schemaIssuerDid": undefined,
            "schemaName": undefined,
            "schemaVersion": undefined,
          },
        ],
      },
    },
    "version": "0.0.1",
  },
}
`;
